### PR TITLE
fix: change switch...case error handling

### PR DESCRIPTION
**Pull request by ivyoverflow**
#13 is closed.

### CODE REVIEW

File: book/internal/handler/book.go [hunk 1]
```diff
@@ -44,13 +44,13 @@ func (h *Book) Insert(rw http.ResponseWriter, r *http.Request) {
...
-		switch {
-		case err == types.ErrorDuplicateValue:
+		switch err {
+		case types.ErrorDuplicateValue:
 			rw.WriteHeader(http.StatusConflict)
 			fmt.Fprintf(rw, `{"error": {"statusCode": %d, "message": "%s"}}`, http.StatusConflict, types.ErrorDuplicateValue.Error())
 
 			return
-		case err == types.ErrorBadRequest:
+		case types.ErrorBadRequest:
```
Comment (kiryalovik) [inline book/internal/handler/book.go:53]: What `ErrorBadRequest` means for service? Does the service have a `request`?

File: book/internal/repository/mongo/book.go [hunk 2]
```diff
@@ -49,8 +49,8 @@ func (r *BookRepository) Get(ctx context.Context, bookID uuid.UUID) (*model.Book
 	receivedBook := model.Book{}
 	err := r.Collection().FindOne(ctx, filter).Decode(&receivedBook)
 	if err != nil {
-		switch {
-		case err == mongo.ErrNoDocuments:
+		switch err {
+		case mongo.ErrNoDocuments:
 			return nil, types.ErrorNotFound
 		default:
 			return nil, types.ErrorInternalServerError
```
Comment (kiryalovik) [inline book/internal/repository/mongo/book.go:56]: Do not replace the original error.  Now you don't know what happened and why you got an error on repo level, cause you replaced.

**Review by kiryalovik — COMMENTED** [review]
Looks good, minor comments

File: book/internal/handler/book.go (hunk 1 above)
Comment (ivyoverflow) [reply book/internal/handler/book.go]: Yeah, I realized my mistake and uploaded the commit with the changes. Thanks.

File: book/internal/repository/mongo/book.go (hunk 2 above)
Comment (ivyoverflow) [reply book/internal/repository/mongo/book.go]: You can see the changes [here](https://github.com/ivyoverflow/pub-sub/pull/17/commits/b5241e94c6daa36f8a3c35023f3d52d0addb88ba).

**Review by kiryalovik — APPROVED** [review]
LGTM.
Resolve conflicts, and do `go mod tidy`

**Pull request merged**